src/audio/copier/copier.c

```diff
@@ -746,7 +748,17 @@ static int copier_prepare(struct comp_dev *dev)
 		return PPL_STATUS_PATH_STOP;
 
 	for (i = 0; i < cd->endpoint_num; i++) {
-		ret = cd->endpoint[i]->drv->ops.prepare(cd->endpoint[i]);
+		if (dev->ipc_config.type == SOF_COMP_HOST) {
+			ret = comp_set_state(cd->endpoint[i], COMP_TRIGGER_PREPARE);
+			if (ret < 0)
+				return ret;
+			if (ret == COMP_STATUS_STATE_ALREADY_SET)
+				return PPL_STATUS_PATH_STOP;
+
+			ret = host_prepare_dma(cd->hd);
+		} else {
+			ret = cd->endpoint[i]->drv->ops.prepare(cd->endpoint[i]);
+		}
```

Comment:
Hmm, this bit now looks a bit ouf-of-place in this PR. There is no gain to open-code the host_prepare() here and we could just call ops.prepare(). So maybe have this part in follow-up PRs and only have the host_copier.h addition and changes to host-legacy.c and host-zephyr.c in this PR.

Not 100% sure of this.

Reply:
@ranj063 any comment here.
@btian1 if this is needed, we need to use the ops wrapper APIs and not directly deference the func ptrs

Reply:
Can we have an inline comment for this new block so we can all understand what it's doing. The
```c
cd->endpoint[i]->drv->ops.prepare(cd->endpoint[i]);
```
Should use the API wrapper to access prepare()

Reply:
@lgirdwood , I am not clear the meaning of "API wrapper", could you give me more hints?